Callback continuation can block pipeline execution

timbussmann opened the pull request.

bording commented on src/NServiceBus.Callbacks/TaskCompletionSourceAdapter.cs:
```diff
@@ -26,12 +26,20 @@ public TaskCompletionSourceAdapter(TaskCompletionSource<TResult> tcs)
         public void TrySetResult(object result)
         {
-            taskCompletionSource.TrySetResult((TResult) result);
+#pragma warning disable PCR0002 // Await or Capture Tasks
```
Would it be better to introduce an [Ignore](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/TaskEx.cs#L14) instead of disabling the warning?

danielmarbach replied:
I agree with @bording

timbussmann replied:
fine with that. I was trying to avoid introducing more code which will be introduced on every patched version. Shouldn't `Ignore` be part of the analyzer itself?

timbussmann commented:
Seems like the currently failing builds are related to something on the build process and not the PR itself.

timbussmann commented:
rebased and squashed the PR. The failing TC builds are related to the PR being merged against develop instead to master on TC for the test build according to @dbelcham so we need to ignore those tests and merge the PR anyway.

@Particular/nservicebus-maintainers please review
